### DISCUSSION TITLE
Auto start the Postal on server reboot without human intervention

### DISCUSSION THREAD

**Discussion by XenyFord** [postalserver/postal#1562] [Help with using Postal]
Hi There!

Can anyone help me with the solution to the problem I've been facing

When I use two different commands in crontab as mentioned below to be executed on server reboot*Ubuntu 21.04), postal is not properly started, as one of the postal elements(postal_worker_1) was not started.

@reboot iptables-restore < /xyz/iptables/xyg.conf
@reboot postal start

So, please help me with a perfect way(command) to make these two lines executed properly without any human intervention on server reboot..

A heartfelt thanks in advance!

**Comment by willpower232** [comment]
The worker was probably waiting for rabbitmq and/or the database to be available. How are you running them?

**Reply by XenyFord** [reply]
Mr. Will,

I'm apologizing for my delayed response, because I was away the town for the past few days..

Coming to the issue: 
I'm running the below two commands within the crontab to make the postal auto start on reboot without human intervention; all the elements as getting started automatically on reboot except "postal_worker_1"

@reboot iptables-restore < /xyz/iptables/xyg.conf
@reboot postal start

Below are the images for your reference
![ERROR](https://user-images.githubusercontent.com/89290310/131736521-be8501d7-9cdb-4279-90a8-e35765264e20.jpg)
![manual-start](https://user-images.githubusercontent.com/89290310/131736533-21f7c420-4eb2-426a-bcbb-b256c1a89ef9.jpg)
...

Thank you in advance!

**Reply by willpower232** [reply]
if you run `postal logs worker` then you can see what problem it had

**Reply by XenyFord** [reply]
Mr. Will - Thank you very much for you prompt response.
The logs are as shown in the image below
![ERROR](https://user-images.githubusercontent.com/89290310/131847642-986d6433-117b-44f5-a2bc-9e334c1c2fbb.jpg)
....

**Reply by willpower232** [reply]
You can see that it hasn't been able to connect to your rabbitmq server. Does it work if you do `postal dc restart worker`

**Reply by XenyFord** [reply]
Yes! Mr. Will, It's working fine when I manually run `postal dc restart worker` or `postal start`, but I need the entire postal server to be auto-started on reboot without any human intervention. Please help me with the exact command that can make me achieve.

Thank you in advance!

![postal dc restart worker](https://user-images.githubusercontent.com/89290310/131853827-33813f06-ed8d-4b94-a2ab-1a67c8bcd8bf.jpg)

**Reply by willpower232** [reply]
that confirms that everything is set up fine, it just needs to start in the right order.

This is already catered for but not in the released installer because it will be down to how each user is configuring their environment

https://github.com/postalserver/postal/blob/2e7aaaba5a9a37f32bdff9a9fde0379834b2a4db/docker-compose.yml#L15-L17

I think you should be able to copy that into the docker-compose.yml file you have and replace the names `mariadb` and `rabbitmq` with `127.0.0.1` and that should do the trick.

**Reply by XenyFord** [reply]
Mr. Will,

If you are not bothered, would you please be so kind to let me know the `exact path` and `command lines` or `exact code` to replace? So I will do it, please...
Also, are the following commands I have in Crontab‌ correct? - my server on Ubuntu 21.04.

`@reboot postal start`
`@reboot iptables-restore < /xyz/iptables/xyg.conf`

Thank you in advance!

**Reply by willpower232** [reply]
I think you'll have a file at /opt/postal/install/docker-compose.yml (not the one in the templates folder), after the section starting with `worker:` you need to add this

```
    environment:
      WAIT_FOR_TIMEOUT: 90
      WAIT_FOR_TARGETS: |-
        127.0.0.1:5672
```

then `postal stop` and `postal start` to apply it and I think that should be enough.

**Reply by XenyFord** [reply]
Mr. Will,
It's stopped the entire postal once I added the below lines into `/opt/postal/install/docker-compose.yml`.

`environment:`
      `WAIT_FOR_TIMEOUT: 90`
      `WAIT_FOR_TARGETS: |-`
       `127.0.0.1:5672`
       
![added](https://user-images.githubusercontent.com/89290310/131975664-dc6b3528-0863-4631-835a-11da191a5ab1.jpg)

And the error is as shown in the image below after I run `postal stop` , `postal start` and `postal status` . 

![error](https://user-images.githubusercontent.com/89290310/131975648-9e43c5a4-d57f-4bf8-b9b6-09d135efffc4.jpg)

I hope you have not misunderstood my requirement. I wanted to have the postal to be started automatically on reboot without any human intervention.

Thank you in advance!

**Reply by willpower232** [reply]
you need to add spaces so that the lines you added are part of worker and not their own section

**Reply by solracsf** [reply]
Same issue here:

```
  worker:
    image: ghcr.io/postalserver/postal:2.0.0
    command: postal worker
    network_mode: host
    volumes:
      - /opt/postal/config:/config
    environment:
      WAIT_FOR_TIMEOUT: 90
      WAIT_FOR_TARGETS: |-
        127.0.0.1:5672
```

But after reboot:

```
# postal status
Name   Command   State   Ports
------------------------------
```

So i need to manually:

```
# postal start
Creating postal_web_1      ... done
Creating postal_worker_1   ... done
Creating postal_cron_1     ... done
Creating postal_smtp_1     ... done
Creating postal_requeuer_1 ... done
```

**Reply by willpower232** [reply]
> then postal stop and postal start to apply it and I think that should be enough.

**Comment by Ascar699** [comment]
Hello, everybody.

I am wondering whether this issue has been resolved?  In plain language - is it possible to make Postal start on machine startup?

Thank you

**Reply by trebla** [reply]
Probably not the best solution but it works:

I'm calling script via cronjob every 2 minutes:
*/2 * * * * ~/postalCheck.sh

---- postalCheck.sh----
#!/bin/bash
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
LD_LIBRARY_PATH=/usr/local/lib
/usr/bin/postal status | grep Exit && date >> ~/logpostal.log
/usr/bin/postal status | grep Exit && /usr/bin/postal start

**Reply by willpower232** [reply]
brutal but easy to diagnose I guess :sweat_smile:
